lang/rust/src/types.rs

```diff
@@ -168,7 +173,11 @@ where
     T: Into<Self>,
 {
     fn from(value: Option<T>) -> Self {
-        Self::Union(Box::new(value.map_or_else(|| Self::Null, Into::into)))
+        // NOTE: this is incorrect in case first type in union is not "none"
```

Comment:
This seems incorrect also when the union has more than two items.
And looks like a stopper for this PR.

Reply:
Ignore this comment! I misunderstood it the first time!
It looks OK to me!